### ISSUE TITLE
WixUI accessibility bugs

### ISSUE THREAD

**Issue by barnson** [wixtoolset/issues#6221]
This is a meta-issue to capture the various accessibility bugs in the WixUI dialog library provided by WixUIExtension. Please add issue numbers here. Accessibility isn't a core competency of the WiX team, so we're looking for help from experts.

https://github.com/wixtoolset/issues/issues/5418
https://github.com/wixtoolset/issues/issues/5903
https://github.com/wixtoolset/issues/issues/5905
https://github.com/wixtoolset/issues/issues/5961
https://github.com/wixtoolset/issues/issues/5994
https://github.com/wixtoolset/issues/issues/6205
https://github.com/wixtoolset/issues/issues/6218

**Comment by barnson** [comment]
https://github.com/wixtoolset/issues/issues/6263

**Comment by barnson** [comment]
https://github.com/wixtoolset/issues/issues/6747

**Comment by fsteffi** [comment]
hi @barnson  is there any update on these bugs to be fixed?

**Comment by barnson** [comment]
https://github.com/wixtoolset/issues/issues/7397
https://github.com/wixtoolset/issues/issues/7398

**Comment by barnson** [comment]
https://github.com/wixtoolset/issues/issues/5961

**Comment by merriemcgaw** [comment]
Just checking in on this issue - there are new EU accessibility requirements for consumer apps that mean WiX might get impacted by many EU devs needing an accessible installer UI. Feel free to ping me for any details you might need.

**Comment by barnson** [comment]
@merriemcgaw We have two problems:

- We're not a11y experts so the drive-by bug dumps tend to sound like the adults in a Charlie Brown TV show. (I'm sure they make sense to the folks opening the bugs.)
- MSI internal UI is ancient (1999-era) and it isn't apparent how much UI automation support is available. If it's present, it's not documented. (And MSI internal UI isn't extensible in any meaningful way.)

So for the moment we're kinda stuck. The many limitations in internal UI have bumped it way down my personal priority list in favor of bundle applications, which can have modern UI with all the goodies.

**Comment by barnson** [comment]
@merriemcgaw But thanks for commenting and offering to help. It's a refreshing change. :)

**Comment by merriemcgaw** [comment]
@barnson thanks, that's not difficult. Have you tried running [AccessibilityInsights for Windows](https://aka.ms/accessibilityinsights) against your sample UI and any default UI for the burn bundles? That should flag any of the issues that would be raised from compliance stuff at the office and often points to links on how/why etc. I also looked in on msiexec.exe UI and you're right, there's not much we can do, and most often teams are able to be compliant because they have cmd line options. If new issues come up that need details or justification I'm happy to help there.

**Comment by barnson** [comment]
@merriemcgaw Yeah, I've looked at it, though there's still a fair amount of background information that falls into the category of "even a glutton for punishment like a setup expert finds difficult to enjoy." Given the limitations of MSI internal UI, we're not likely to make much progress on these issues. The stock bundle UI is much more under control, so is more likely to get attention (eventually).

**Comment by barnson** [comment]
https://github.com/wixtoolset/issues/issues/7926

**Comment by msftedad** [comment]
Hi @barnson , Is there any update on this issue?
